packages/synapse-react-client/src/utils/hooks/useUploadFileEntity/usePrepareFileEntityUpload.test.ts

```diff
@@ -0,0 +1,171 @@
+import { renderHook as _renderHook } from '@testing-library/react'
+import { MOCK_CONTEXT_VALUE } from '../../../mocks/MockSynapseContext'
+import { getUseMutationMock } from '../../../testutils/ReactQueryMockUtils'
+import { createWrapper } from '../../../testutils/TestingLibraryUtils'
+import * as GetFileEntityIdWithSameNameModule from './getFileEntityIdWithSameName'
+import * as UseCreatePathsAndGetParentIdModule from './useCreatePathsAndGetParentId'
+
+import { usePrepareFileEntityUpload } from './usePrepareFileEntityUpload'
+
+const mockUseCreatePathsAndGetParentId = jest.spyOn(
+  UseCreatePathsAndGetParentIdModule,
+  'useCreatePathsAndGetParentId',
+)
+const mockGetFileEntityIdWithSameName = jest.spyOn(
+  GetFileEntityIdWithSameNameModule,
+  'getFileEntityIdWithSameName',
+)
+
+describe('usePrepareFileEntityUpload', () => {
+  beforeEach(() => {
+    jest.clearAllMocks()
+  })
+
+  function renderHook() {
+    return _renderHook(() => usePrepareFileEntityUpload(), {
+      wrapper: createWrapper(),
+    })
+  }
+  const newFile = new File([''], 'newFile.txt')
+  const updatedFile = new File([''], 'existingFile.txt')
+  const files = [newFile, updatedFile]
+  const parentId = 'syn123'
+  const updatedEntityId = 'syn456'
+
+  test('prepares a new file entity for upload and an existing file entity for update', async () => {
+    const useCreatePathsAndGetParentIdMockResult = getUseMutationMock(null)
+    useCreatePathsAndGetParentIdMockResult.mutateAsync.mockImplementation(
+      ({ file }) => ({ file, parentId }),
+    )
+    mockUseCreatePathsAndGetParentId.mockReturnValue(
+      useCreatePathsAndGetParentIdMockResult,
+    )
+
+    mockGetFileEntityIdWithSameName.mockImplementation(fileName => {
+      if (fileName === 'newFile.txt') {
```

Comment:
Prefer logic to be reversed - look for "existingFile.txt" and return the updatedEntityId, but by default it's a new file